sql/src/main/java/io/crate/execution/ddl/RerouteActions.java

```diff
@@ -86,6 +93,27 @@ private RerouteActions() {
         return listener;
     }
 
+    public static ClusterRerouteRequest prepareAllocateStalePrimaryReq(PromoteReplicaStatement promoteReplica,
+                                                                       Functions functions,
+                                                                       Row parameters,
+                                                                       TransactionContext txnCtx) {
+        String index = RerouteActions.getRerouteIndex(promoteReplica, parameters);
+        Integer shardId = DataTypes.INTEGER.value(SymbolEvaluator.evaluate(
+            txnCtx, functions, promoteReplica.shardId(), parameters, SubQueryResults.EMPTY));
+        if (shardId == null) {
+            throw new NullPointerException("Shard in REROUTE PROMOTE REPLICA must not be null");
+        }
+        Boolean acceptDataLoss = DataTypes.BOOLEAN.value(SymbolEvaluator.evaluate(
+            txnCtx, functions, promoteReplica.acceptDataLoss(), parameters, SubQueryResults.EMPTY));
+        if (acceptDataLoss == null) {
```

Comment:
could this ever be true? as we convert a missing param to FALSE already in the analyzer (should this be an assertion if that's the case?)

Reply:
A user could set it explicitly via `WITH (accept_data_loss = null)` or `WITH (accept_data_loss = ?)` and `args = [null]`. 

The default is only applied if the key is not present, independent of the value. And since parameter values cannot be resolved eagerly in the analyzer the null check can't be there.